dsgrid/config/file_schema.py

```diff
@@ -0,0 +1,221 @@
+import logging
+from pathlib import Path
+from typing import Self
+
+from pydantic import Field, field_validator, model_validator
+
+from dsgrid.data_models import DSGBaseModel
+from dsgrid.dimension.base_models import DimensionType
+from dsgrid.exceptions import DSGInvalidDataset, DSGInvalidField
+from dsgrid.spark.functions import read_csv_duckdb, read_json, read_parquet
+from dsgrid.spark.types import DataFrame
+from dsgrid.utils.utilities import check_uniqueness
+
+
+SUPPORTED_TYPES = set(
+    (
+        "BOOLEAN",
+        "INT",
+        "INTEGER",
+        "TINYINT",
+        "SMALLINT",
+        "BIGINT",
+        "FLOAT",
+        "DOUBLE",
+        "TIMESTAMP_TZ",
+        "TIMESTAMP_NTZ",
+        "STRING",
+        "TEXT",
+        "VARCHAR",
+    )
+)
+
+DUCKDB_COLUMN_TYPES = {
+    "BOOLEAN": "BOOLEAN",
+    "INT": "INTEGER",
+    "INTEGER": "INTEGER",
+    "TINYINT": "TINYINT",
+    "SMALLINT": "INTEGER",
+    "BIGINT": "BIGINT",
+    "FLOAT": "FLOAT",
+    "DOUBLE": "DOUBLE",
+    "TIMESTAMP_TZ": "TIMESTAMP WITH TIME ZONE",
+    "TIMESTAMP_NTZ": "TIMESTAMP",
+    "STRING": "VARCHAR",
+    "TEXT": "VARCHAR",
+    "VARCHAR": "VARCHAR",
+}
+
+SPARK_COLUMN_TYPES = {
+    "BOOLEAN": "BOOLEAN",
+    "INT": "INT",
+    "INTEGER": "INT",
+    "TINYINT": "TINYINT",
+    "SMALLINT": "SMALLINT",
+    "BIGINT": "BIGINT",
+    "FLOAT": "FLOAT",
+    "DOUBLE": "DOUBLE",
+    "STRING": "STRING",
+    "TEXT": "STRING",
+    "VARCHAR": "STRING",
+    "TIMESTAMP_TZ": "TIMESTAMP",
+    "TIMESTAMP_NTZ": "TIMESTAMP_NTZ",
+}
+
+assert sorted(DUCKDB_COLUMN_TYPES.keys()) == sorted(SPARK_COLUMN_TYPES.keys())
+assert not SUPPORTED_TYPES.difference(DUCKDB_COLUMN_TYPES.keys())
+
+logger = logging.getLogger(__name__)
+
+
+class Column(DSGBaseModel):
+    name: str = Field(description="Name of the column")
+    dimension_type: DimensionType | None = Field(
+        default=None,
+        description="Dimension represented by the data in the column. Optional if this is a "
+        "time column or pivoted column. Required if the column represents a stacked dimension "
+        "but an alternate name is being used, such as 'county' instead of 'geography'. "
+        "dsgrid will rename any column that is set at runtime.",
```

Comment:
```suggestion
        "dsgrid will rename any column that is set at runtime, writing out the result to the registry's data directory.",
```

Reply:
Seems helpful to be clear that the renaming won't modify the user's original file? I also thought about adding the words "in memory" somewhere.